core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -520,6 +520,8 @@ object SparkSubmit {
       }
       if (args.principal != null) {
         require(args.keytab != null, "Keytab must be specified when the keytab is specified")
+        sysProps.put("spark.yarn.keytab", args.keytab)
+        sysProps.put("spark.yarn.principal", args.principal)
```

Comment:
@harishreedharan Hi Hari, could you let us know the preferred way to pass principal and keytab  parameters from spark submit to spark sql? waiting for your response to proceed. Thank you!